### ISSUE TITLE
-Werror gcc/4.8.4: unknown-pragmas #pragma omp simd with OpenMP build

### ISSUE THREAD

**Issue by ndellingwood** [kokkos/kokkos-kernels#180]
```
In file included from /home/ndellin/kokkos-kernels/src/blas/impl/KokkosBlas3_gemm_spec.hpp:51:0,
                 from /home/ndellin/kokkos-kernels/src/impl/generated_specializations_cpp/gemm/KokkosBlas3_gemm_eti_spec_inst_double_LayoutLeft_Serial_HostSpace.cpp:46:
/home/ndellin/kokkos-kernels/src/blas/impl/KokkosBlas3_gemm_impl.hpp:343:0: error: ignoring #pragma omp simd [-Werror=unknown-pragmas]
     #pragma omp simd
 ^
In file included from /home/ndellin/kokkos-kernels/src/blas/impl/KokkosBlas3_gemm_spec.hpp:51:0,
                 from /home/ndellin/kokkos-kernels/src/impl/generated_specializations_cpp/gemm/KokkosBlas3_gemm_eti_spec_inst_double_LayoutLeft_Serial_HBWSpace.cpp:46:
/home/ndellin/kokkos-kernels/src/blas/impl/KokkosBlas3_gemm_impl.hpp:343:0: error: ignoring #pragma omp simd [-Werror=unknown-pragmas]
     #pragma omp simd
 ^
In file included from /home/ndellin/kokkos-kernels/src/blas/impl/KokkosBlas3_gemm_spec.hpp:51:0,
                 from /home/ndellin/kokkos-kernels/src/impl/generated_specializations_cpp/gemm/KokkosBlas3_gemm_eti_spec_inst_double_LayoutRight_Serial_HBWSpace.cpp:46:
/home/ndellin/kokkos-kernels/src/blas/impl/KokkosBlas3_gemm_impl.hpp:343:0: error: ignoring #pragma omp simd [-Werror=unknown-pragmas]
     #pragma omp simd
 ^
In file included from /home/ndellin/kokkos-kernels/src/blas/impl/KokkosBlas3_gemm_spec.hpp:51:0,
                 from /home/ndellin/kokkos-kernels/src/impl/generated_specializations_cpp/gemm/KokkosBlas3_gemm_eti_spec_inst_double_LayoutRight_Serial_HostSpace.cpp:46:
/home/ndellin/kokkos-kernels/src/blas/impl/KokkosBlas3_gemm_impl.hpp:343:0: error: ignoring #pragma omp simd [-Werror=unknown-pragmas]
     #pragma omp simd
cc1plus: all warnings being treated as errors
```

**Comment by ndellingwood** [comment]
PR #181 issued with a fix.

**Comment by ndellingwood** [comment]
gcc/4.8.4 implements OpenMP version 3.1 which does not support `simd` constructs (at least nowhere to be found in reference here [openmp3.1](http://www.openmp.org/wp-content/uploads/OpenMP3.1.pdf)), hence the -Werror for unknown pragma. 

Relevant to discussion in trilinos/Trilinos#2317.

**Comment by nmhamster** [comment]
Can we if def those out?

**Comment by ndellingwood** [comment]
@nmhamster yeah, I just put in PR #181 doing that

**Comment by brian-kelley** [comment]
I'm reopening this because while doing the full cm_test_all_sandia on kokkos-dev2, 4.8.4 gives this:
```
 111 In file included from /ascldap/users/bmkelle/StdSortWorkaround-fulltest-1587158944/kokkos-kernels/src/batched/KokkosBatched_Vector.hpp:166:0,
 112                  from /ascldap/users/bmkelle/StdSortWorkaround-fulltest-1587158944/kokkos-kernels/src/batched/KokkosBatched_Trsm_Decl.hpp:8,
 113                  from /ascldap/users/bmkelle/StdSortWorkaround-fulltest-1587158944/kokkos-kernels/src/blas/impl/KokkosBlas3_trsm_impl.hpp:57,
 114                  from /ascldap/users/bmkelle/StdSortWorkaround-fulltest-1587158944/kokkos-kernels/src/blas/impl/KokkosBlas3_trsm_spec.hpp:53,
 115                  from /ascldap/users/bmkelle/StdSortWorkaround-fulltest-1587158944/Testing/TestAll_2020-04-17_14.29.10/gcc/4.8.4/OpenMP-release/src/impl/generated_specia     lizations_cpp/trsm/Blas3_trsm_eti_DOUBLE_LAYOUTLEFT_EXECSPACE_OPENMP_MEMSPACE_HOSTSPACE.cpp:48:
 116 /ascldap/users/bmkelle/StdSortWorkaround-fulltest-1587158944/kokkos-kernels/src/batched/KokkosBatched_Vector_SIMD.hpp:49:0: error: ignoring #pragma omp simd [-Werror=unk     nown-pragmas]
 117  #pragma omp simd
 118  ^
 119 /ascldap/users/bmkelle/StdSortWorkaround-fulltest-1587158944/kokkos-kernels/src/batched/KokkosBatched_Vector_SIMD.hpp:63:0: error: ignoring #pragma omp simd [-Werror=unk     nown-pragmas]
 120  #pragma omp simd
 121  ^
 122 /ascldap/users/bmkelle/StdSortWorkaround-fulltest-1587158944/kokkos-kernels/src/batched/KokkosBatched_Vector_SIMD.hpp:77:0: error: ignoring #pragma omp simd [-Werror=unk     nown-pragmas]
 123  #pragma omp simd
 124  ^
 125 /ascldap/users/bmkelle/StdSortWorkaround-fulltest-1587158944/kokkos-kernels/src/batched/KokkosBatched_Vector_SIMD.hpp:92:0: error: ignoring #pragma omp simd [-Werror=unk     nown-pragmas]
 126  #pragma omp simd
 127  ^
 128 /ascldap/users/bmkelle/StdSortWorkaround-fulltest-1587158944/kokkos-kernels/src/batched/KokkosBatched_Vector_SIMD.hpp:113:0: error: ignoring #pragma omp simd [-Werror=un     known-pragmas]
 129  #pragma omp simd
 130  ^
```
etc.

**Comment by brian-kelley** [comment]
The new fix for this is in #698, but other things in the PR aren't ready for merging yet.

**Comment by brian-kelley** [comment]
Fixd again in 3.1.1.
